factory/api/apiResolverFactory.go

```diff
@@ -234,6 +234,7 @@ func CreateApiResolver(args *ApiResolverArgs) (facade.ApiResolver, error) {
 		TxTypeHandler:            txTypeHandler,
 		LogsFacade:               logsFacade,
 		DataFieldParser:          dataFieldParser,
+		ChainID:                  args.Configs.GeneralConfig.GeneralSettings.ChainID,
```

Comment:
I honestly didn't know it can change to a different value during on-chain execution, but not sure where else to get this field from(for all tx types).
Any suggestion? Otherwise, would leave it as it is

Reply:
do you actually need the ChainID here?